### ISSUE TITLE
RPI 1640: PARM doco incorrect

### ISSUE THREAD

**Issue by adelosa** [z390development/z390#210]
Raised by Hugh Sweeney 2019/08/01
Documentation on PARM is incorrect in Program Services Guide and in z390_options.htm. It omits the fullword in storage in between the register and the parm data. See mail by Hugh Sweeney dd 2019-07-14 16:51

**Comment by z390org** [comment]
Email by Hugh Sweeney:


Hello from sunny Cardiff,

The page at  http://www.z390.org/z390_Options.htm  says:

    Define character string parm to passed to executable program via
    address in R1 at startup pointing to half word length followed by EBCDIC
    characters. Default length is 0.


but when I coded that I found the halfword length field was always zero, whether or not a PARM was used on the command line. The phrase "address in R1 at startup pointing to half word length" should read "address in R1 at startup pointing to a fullword pointing to half word length". (Also typo:"be" should appear before "passed".)

As it happens, that fullword address in storage always had its upper 16 bits all zero, and that's what I was takng as the length as per the documentation.

Am I the first person to use the PARM option, or is this a z/OS navigation convention well known to everyone but me so they all ignore the documentation?

I find the sentence "Default length is 0" unambiguous but unhelpful. If it means "when the length is zero, no PARM is present" I'd prefer it said something like that.

Hugh
--
